stream.go

```diff
@@ -409,26 +398,17 @@ func (s *Stream) readData(hdr header, flags uint16, conn io.Reader) error {
 	// Wrap in a limited reader
 	conn = &io.LimitedReader{R: conn, N: int64(length)}
```

Comment:
Actually, I think we can:

1. Drop this.
2. Modify the receive buffer to stop when we've read the full length.

This is technically correct, just a bit weird.